modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/MetadataKeysDelegate.java

```diff
@@ -85,11 +87,11 @@
     try {
       final Map<Integer, ParameterModel> partsByOrder = getPartOrderMapping(keyParts);
       Set<MetadataKey> metadataKeys;
-      if (keyResolver instanceof PartialTypeKeysResolver && hasInitialLevel(partialKey, partsByOrder, reflectionCache)) {
-        metadataKeys = singleton(((PartialTypeKeysResolver) keyResolver).resolveChilds(context, partialKey));
-      } else {
-        metadataKeys = keyResolver.getKeys(context);
-      }
+
+      metadataKeys = context instanceof ConnectionProviderAwareMetadataContext
```

Comment:
Unify with the definition of the variable
`Set<MetadataKey> metadataKeys = context instanceOf ConnectionProviderAwareMetadaContext...`